packages/client-react-streaming/src/ManualDataTransport/serialization.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Stringifies a value to be injected into JavaScript "text" - preverves `undefined` values.
+ */
+export function stringify(value: any) {
+  let undefinedPlaceholder = "$apollo.undefined$";
+
+  const stringified = JSON.stringify(value);
+  while (stringified.includes(JSON.stringify(undefinedPlaceholder))) {
+    undefinedPlaceholder = "$" + undefinedPlaceholder;
+  }
+  return JSON.stringify(value, (_, v) =>
+    v === undefined ? undefinedPlaceholder : v
+  ).replaceAll(JSON.stringify(undefinedPlaceholder), "undefined");
+}
+
+export function revive(value: any): any {
+  return value;
```

Comment:
I see this `revive` function replaced the `SuperJSON.deserialize` call. 

I'm curious about the default here to just return the value unchanged instead of trying to parse it. Should this do a `JSON.parse` or equivalent of some kind?

Reply:
Before, we had `SuperJSON.stringify` and `SuperJSON.deserialize` here.

Quoting their docs:

* serialize: Serializes any JavaScript value into a JSON-compatible object.
* deserialize: Deserializes the output of Superjson back into your original value.
* stringify: Serializes and then stringifies your JavaScript value.
* parse: Parses and then deserializes the JSON string returned by stringify.

=> 
* in one direction, we need to serialize and stringify
* in the other direction, parsing is done by the browser and we only need to deserialize (if there was any additionaly serialization done in the first place)